### PR TITLE
Refactorings now that we know more Elixir

### PR DESCRIPTION
**Pull request by michaelgundlach**
Hey, can you give michaelgundlach push rights to nathanl/wordular?

### CODE REVIEW

File: web/models/anagrams.ex [hunk 1]
```diff
@@ -30,30 +28,17 @@ defmodule Anagrams do
...
-    entry_letterbag = letterbag(current_word)
-    if Map.has_key?(output_dict, entry_letterbag) do
-      existing_entry = Map.get(output_dict, entry_letterbag)
-      updated_dict = Map.put(output_dict, entry_letterbag, existing_entry ++ [current_word])
-    else
-      updated_dict = Map.put(output_dict, entry_letterbag, [current_word])
-    end
-    dictionary(remaining_words, updated_dict)
+  def dictionary(human_readable_dictionary) do
+    Enum.reduce(human_readable_dictionary, %{}, fn word, map_acc ->
```
Comment (nathanl) [inline web/models/anagrams.ex:33]: Wow, you really _reduced_ the amount of code here. :trollface:

Comment (nathanl) [inline web/models/anagrams.ex:33]: Seriously, my brain is too tired to comprehend this right now but I'd like you to walk me through it.

File: web/models/anagrams.ex [hunk 2]
```diff
@@ -118,11 +102,15 @@ defmodule Anagrams do
...
 
   def usable_entries_for(dict_entries, phrase) do
-    for entry <- dict_entries, subtractable_from?(entry, phrase), do: entry
+    Enum.filter(dict_entries, &(subtractable_from?(&1, phrase)))
   end
 
   def subtractable_from?(needles, haystack) do
     thing = haystack |> without(needles)
+    # TODO: Enum.count(a_list) is O(n).  If the letterbags we passed around
+    # were not sorted lists but a %Letterbag{sorted_letters: [...], size: 48}
```
Comment (nathanl) [inline web/models/anagrams.ex:111]: Oh, I like the idea of making a Letterbag struct.

**Comment by nathanl** [comment]
Yep, you have push access now.

**Comment by michaelgundlach** [comment]
you up?

**Pull request merged**